### PR TITLE
Spec version filter for cli init

### PR DESCRIPTION
**Pull request by wqsz7xn**
Adds a filter of templates for users chosen specVersion. If there are no templates for that specVersion (which should not be the case) then it falls back to a custom git remote & branch.

### CODE REVIEW

File: packages/cli/src/commands/init.ts [hunk 1]
```diff
@@ -113,65 +113,75 @@ export default class Init extends Command {
...
-      const paddingWidth = candidateTemplates.map(({name}) => name.length).reduce((acc, xs) => Math.max(acc, xs)) + 5;
-
-      const templateDisplays = candidateTemplates.map(
-        ({description, name}) => `${name.padEnd(paddingWidth, ' ')}${chalk.gray(description)}`
-      );
-      templateDisplays.push(`${'Other'.padEnd(paddingWidth, ' ')}${chalk.gray('Enter a custom git endpoint')}`);
+      const networks = templates
+        .map(({network}) => network)
+        .filter((n, i, self) => {
+          return i === self.indexOf(n);
```
Comment (ianhe8x) [inline packages/cli/src/commands/init.ts:126]: is this a trick to do unique? I'm not too confident with it, can you just use unique from lodash?

File: packages/cli/src/commands/init.ts [hunk 2]
```diff
@@ -113,65 +113,75 @@ export default class Init extends Command {
...
+        });
+      networks.push('Other');
 
+      // Network
+      inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
```
Comment (ianhe8x) [inline packages/cli/src/commands/init.ts:131]: would be better the setup steps be moved to the beginning of the file, (right after imports).

File: packages/cli/src/commands/init.ts (hunk 1 above)
Comment (wqsz7xn) [reply packages/cli/src/commands/init.ts:126]: Sure

**Comment by wqsz7xn** [comment]
Let me put back in the `--starter` flag before merging this. I don't know where it disappeared. My bad

**Comment by wqsz7xn** [comment]
Added back the `--starter` flag so that I don't have to change a bunch of docs

**Comment by ianhe8x** [comment]
The template feature is a replacement for the original --starter. With --starter, it skips template selection. I think we should move on without --starter

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**